linkcheck/url.py

```diff
@@ -395,10 +397,10 @@ def url_quote (url, encoding):
 
 def document_quote (document):
     """Quote given document."""
-    doc, query = urllib.parse.splitquery(document)
-    doc = urllib.parse.quote(doc, safe='/=,')
-    if query:
-        return "%s?%s" % (doc, query)
+    o = urllib.parse.urlparse(document)
+    doc = urllib.parse.quote(o.path, safe='/=,')
+    if o.query:
+        return "%s?%s" % (doc, o.query)
```

Comment:
This seems fine, as long as no part of the `document` contains characters that could be misinterpreted, such as : or @ or //.

I'd have to find all callers of document_quote and see where the `document` arg comes from, to be thorough.  I don't want to...